### PR TITLE
Bug fixes for Backend JWT generation

### PR DESCRIPTION
**Pull request by VirajSalaka**
### Issues

Fixes https://github.com/wso2/product-microgateway/issues/2687

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/standalone/jwtGenerator/JwtGeneratorTestCase.java [hunk 1]
```diff
@@ -78,5 +79,55 @@ public void testResponseJWTGenerationToken() throws Exception {
...
         Assert.assertEquals(tokenBody.get("keytype"), TestConstant.KEY_TYPE_PRODUCTION,
                 "Key type is not set correctly in JWT generator");
+        Assert.assertNotNull(tokenBody.get("iat"));
+        Assert.assertNotNull(tokenBody.get("exp"));
+
+        long expValue =  Long.parseLong(String.valueOf(tokenBody.get("exp")));
+        long iatValue =  Long.parseLong(String.valueOf(tokenBody.get("iat")));
+
+        long currentTime = TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis());
+        Assert.assertEquals(expValue - currentTime, 3600, "Time duration between iat and exp claims is " +
```
Comment (pubudu538) [inline integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/standalone/jwtGenerator/JwtGeneratorTestCase.java:89]: We should subtract iat from exp.

Comment (VajiraPrabuddhaka) [inline integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/standalone/jwtGenerator/JwtGeneratorTestCase.java:89]:
```suggestion
        Assert.assertEquals(expValue - iatValue, 3600, "Time duration between iat and exp claims is " +
```

File: integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/standalone/jwtGenerator/JwtGeneratorTestCase.java [hunk 2]
```diff
@@ -78,5 +79,55 @@ public void testResponseJWTGenerationToken() throws Exception {
...
+        headers.put(HttpHeaderNames.AUTHORIZATION.toString(), "Bearer " + newJwtToken);
+        response = HttpsClientRequest.doGet(Utils.getServiceURLHttps(
+                "/v2/standard/jwttoken") , headers);
+        Assert.assertNotNull(response);
+        Assert.assertEquals(response.getResponseCode(), 200, "Response code mismatched");
+
+        responseJSON = new JSONObject(response.getData());
+        String token3 = responseJSON.get("token").toString();
+
+        Assert.assertEquals(token1, token3, "Generated Backend JWTs for two invocations from the different " +
```
Comment (VajiraPrabuddhaka) [inline integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/standalone/jwtGenerator/JwtGeneratorTestCase.java:129]:
```suggestion
        Assert.assertNotEquals(token1, token3, "Generated Backend JWTs for two invocations from the different " +
```

File: integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/standalone/jwtGenerator/JwtGeneratorTestCase.java [hunk 3]
```diff
@@ -78,5 +79,55 @@ public void testResponseJWTGenerationToken() throws Exception {
...
+        Assert.assertNotNull(tokenBody.get("exp"));
+
+        long expValue =  Long.parseLong(String.valueOf(tokenBody.get("exp")));
+        long iatValue =  Long.parseLong(String.valueOf(tokenBody.get("iat")));
+
+        long currentTime = TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis());
+        Assert.assertEquals(expValue - iatValue, 3600, "Time duration between iat and exp claims is " +
+                "not 3600 seconds.");
+        Assert.assertTrue(expValue > currentTime, "Expiry time is not greater than currentTime.");
+        Assert.assertTrue(iatValue < currentTime, "IAT value is not less than the current Time");
```
Comment (VajiraPrabuddhaka) [inline integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/standalone/jwtGenerator/JwtGeneratorTestCase.java:92]:
```suggestion
        Assert.assertTrue(iatValue <= currentTime, "IAT value is not less than the current Time");
```

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : dev-deployment-v2 : 20220711.20

**Comment by choreo-cicd** [comment]
[succeeded] DataplaneE2 cluster : dev-deployment-v2 : 20220711.20

**Comment by choreo-cicd** [comment]
[succeeded] DataplaneE1 cluster : prod-deployment-v2 : 20220719.6

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : prod-deployment-v2 : 20220719.6

**Pull request merged**